src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java

```diff
@@ -0,0 +1,68 @@
+package school.hei.patrimoine.compiler;
+
+import static javax.tools.ToolProvider.getSystemJavaCompiler;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.function.Function;
+import lombok.SneakyThrows;
+
+public class CasFileCompiler implements Function<String, Class<?>> {
+
+  private static final String COMPILE_DIR_NAME =
+      System.getProperty("user.home") + "/.patrimoine/compile";
+  public static final String DEPENDENCY_JAR_PATH =
+      System.getProperty("user.home") + "/Downloads/drive";
```

Comment:
to ensure the directories you will create allow READ and WRITE, use /temp. 
There's a dedicated Java utility which gets temp dir of system

Reply:
done